sphinx/writers/manpage.py

```diff
@@ -306,15 +305,23 @@ def visit_image(self, node: Element) -> None:
         self.body.append(_('[image]') + '\n')
         raise nodes.SkipNode
 
+
+    # Signal our support for this feature - clients can probe for this
+    # attribute to determine if we will emit OSC 8 markup.
+    HAVE_GROFF_OSC_8 = object()
+
     # overwritten -- don't visit inner marked up nodes
     def visit_reference(self, node: Element) -> None:
+        uri = node.get('refuri', '')
+        # Emit OSC 8 markup via the groff device control command.
+        self.body.append(fr"\X'tty: link {uri}'")
```

Comment:
Should you really emit the link if 'uri' is empty?

Reply:
I guess we could skip that but maybe this is a case of GIGO.
I'm not aware of a way to produce a link with an empty URL -- ``` `foo<>`__ ``` doesn't seem to work.
So I'm not sure if we're really helping anyone by masking a potential bug.
That being said, it doesn't really matter and I trust you to make the better call here.

Reply:
Actually, I was asking because I didn't know either! just wondering but what would happen if *uri* is empty? like, would the output be bad? or just the link non-clickable?

Reply:
In my terminal, the first two are clickable (actually the terminal has a "hint mode" where you can copy URLs to clipboard) but the last empty link is not clickable indeed.

```
$ printf '\e]8;;http://example.com\e\\This is a link\e]8;;\e\\\\\n'
This is a link
$ printf '\e]8;;foo\e\\This is a link\e]8;;\e\\\\\n'
This is a link
$ printf '\e]8;;\e\\This is a link\e]8;;\e\\\\\n'
This is a link
```

I don't think it's specified what to do with empty links

Reply:
Mmh... let's do nothing then !